docs/shared/guides/remix.md

```diff
@@ -0,0 +1,181 @@
+⚠️ REMIX FUTURE CHANGE: The route file convention is changing in v2. You can prepare for this change at your convenience with the `v2_routeConvention` future flag. For instructions on making this change see https://remix.run/docs/en/v1.15.0/pages/v2#file-system-route-convention
+💿 Building...
+💿 Rebuilt in 377ms
+Remix App Server started at http://localhost:3000 (http://192.168.0.14:3000)
+```
+
+3. To test the application using vitest run:
+
+```{% command="nx test myapp" path="~/acme" %}
+> nx run myapp:test
+
+ RUN  v0.31.4 /Users/columferry/dev/nrwl/issues/remixguide/acme/apps/myapp
+stderr | app/routes/index.spec.ts > test > should render
+Warning: Functions are not valid as a React child. This may happen if you return a Component instead of <Component /> from render. Or maybe you meant to call this function rather than return it.
+ ✓ app/routes/index.spec.ts  (1 test) 10ms
+ Test Files  1 passed (1)
+      Tests  1 passed (1)
+   Start at  16:15:45
+   Duration  1.20s (transform 51ms, setup 139ms, collect 180ms, tests 10ms, environment 379ms, prepare 103ms)
+
+ ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target test for project myapp (2s)
+```
+
+## Generating an Nx Library
+
+When developing your application, it often makes sense to split your codebase into smaller more focused libraries.
+
+To generate a library to use in your Remix application run:
+
+```{% command="nx g @nx/remix:lib login" path="~/acme" %}
+>  NX  Generating @nx/remix:library
+
+✔ What test runner should be used? · vitest
+UPDATE nx.json
+UPDATE package.json
+CREATE babel.config.json
+CREATE libs/login/project.json
+CREATE libs/login/.eslintrc.json
+CREATE libs/login/README.md
+CREATE libs/login/src/index.ts
+CREATE libs/login/tsconfig.lib.json
+CREATE libs/login/tsconfig.json
+CREATE libs/login/vite.config.ts
+CREATE libs/login/tsconfig.spec.json
+CREATE libs/login/src/lib/login.module.css
+CREATE libs/login/src/lib/login.spec.tsx
+CREATE libs/login/src/lib/login.tsx
+UPDATE tsconfig.base.json
+CREATE libs/login/src/test-setup.ts
+CREATE libs/login/src/server.ts
+```
+
+You can then use the library by importing one of the exports into your application:
+
+`apps/myapp/app/routes/index.tsx`
+
+```tsx
+import { Login } from '@acme/login';
+
+export default function Index() {
+  return (
+    <div>
+      <Login />
+    </div>
+  );
+}
+```
+
+You can also run test on your library:
+
+`nx test login`
+
+## GitHub Repository with Example
+
+You can see an example of an Nx Workspace using Remix by clicking below.
+
+{% github-repository url="https://github.com/Coly010/nx-remix-example" /%}
```

Comment:
maybe should be an `nrwl/nx-recipes` repo instead?

Reply:
It will also be nice to include a route that uses a library for loader/action/component.